app/assets/javascripts/pretty-text/addon/engines/discourse-markdown-it.js

```diff
@@ -578,14 +579,16 @@ export function setup(opts, siteSettings, state) {
 
 export function cook(raw, opts) {
   // we still have to hoist html_raw nodes so they bypass the allowlister
-  // this is the case for oneboxes
-  let hoisted = {};
-  opts.discourse.hoisted = hoisted;
+  // this is the case for oneboxes and also certain plugins that require
+  // raw HTML rendering within markdown bbcode rules
+  opts.discourse.hoisted = opts.discourse.hoisted ??= {};
```

Comment:
This line looks slightly odd to me:

```suggestion
  if (!opts.discourse.hoisted) {
    opts.discourse.hoisted = {};
  }
```

Reply:
Oh actually I wrote this wrong, it should be ` opts.discourse.hoisted ??= {};`, will fix. Its https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment and perfectly valid, we already use it in other places in core